sdk/test/trace/probability_sampler_test.cc

```diff
@@ -0,0 +1,267 @@
+	opentelemetry::trace::TraceId valid_trace_id(buf);
+
+	sampling_result = s1.ShouldSample(nullptr, valid_trace_id, "", span_kind, view);
+
+	ASSERT_EQ(Decision::NOT_RECORD, sampling_result.decision);
+	ASSERT_EQ(nullptr, sampling_result.attributes);
+
+	ProbabilitySampler s2(0.50000001);
+
+	sampling_result = s2.ShouldSample(nullptr, valid_trace_id, "", span_kind, view);
+
+	ASSERT_EQ(Decision::RECORD_AND_SAMPLE, sampling_result.decision);
+	ASSERT_EQ(nullptr, sampling_result.attributes);
+
+	ProbabilitySampler s3(0.49999999);
+
+	sampling_result = s3.ShouldSample(nullptr, valid_trace_id, "", span_kind, view);
+
+	ASSERT_EQ(Decision::NOT_RECORD, sampling_result.decision);
+	ASSERT_EQ(nullptr, sampling_result.attributes);
+
+	ProbabilitySampler s4(0.50000000);
+
+	sampling_result = s4.ShouldSample(nullptr, valid_trace_id, "", span_kind, view);
+
+	ASSERT_EQ(Decision::RECORD_AND_SAMPLE, sampling_result.decision);
+	ASSERT_EQ(nullptr, sampling_result.attributes);
+}
+
+TEST(ProbabilitySampler, ShouldSampleWithContext)
+{
+	opentelemetry::trace::TraceId trace_id;
+	opentelemetry::trace::SpanKind span_kind = opentelemetry::trace::SpanKind::kInternal;
+	opentelemetry::sdk::trace::Sampler::SpanContext c1(false, false);
+	opentelemetry::sdk::trace::Sampler::SpanContext c2(true, false);
+	opentelemetry::sdk::trace::Sampler::SpanContext c3(false, true);
+	opentelemetry::sdk::trace::Sampler::SpanContext c4(true, true);
+
+	using M = std::map<std::string, int>;
+	M m1 = {{}};
+	opentelemetry::trace::KeyValueIterableView<M> view{m1};
+
+ 	ProbabilitySampler s1(0.01);
+
+	auto sampling_result = s1.ShouldSample(&c1, trace_id, "", span_kind, view);
+
+	ASSERT_EQ(Decision::NOT_RECORD, sampling_result.decision);
+	ASSERT_EQ(nullptr, sampling_result.attributes);
+
+	sampling_result = s1.ShouldSample(&c2, trace_id, "", span_kind, view);
+
+	ASSERT_EQ(Decision::RECORD_AND_SAMPLE, sampling_result.decision);
+	ASSERT_EQ(nullptr, sampling_result.attributes);
+
+	sampling_result = s1.ShouldSample(&c3, trace_id, "", span_kind, view);
+
+	ASSERT_EQ(Decision::RECORD_AND_SAMPLE, sampling_result.decision);
+	ASSERT_EQ(nullptr, sampling_result.attributes);
+
+	sampling_result = s1.ShouldSample(&c4, trace_id, "", span_kind, view);
+
+	ASSERT_EQ(Decision::RECORD_AND_SAMPLE, sampling_result.decision);
+	ASSERT_EQ(nullptr, sampling_result.attributes);
+}
+
+TEST(ProbabilitySampler, ProbabilitySamplerHalf)
+{
+	double probability = 0.001;
+	int iterations = 100000, expected_count = iterations * probability,
+		actual_count = 0, variance = 1000;
+
+	opentelemetry::sdk::trace::Sampler::SpanContext c(true, true);
+	ProbabilitySampler s(probability);
+	opentelemetry::trace::SpanKind span_kind = opentelemetry::trace::SpanKind::kInternal;
+
+	using M = std::map<std::string, int>;
+	M m1 = {{}};
+	opentelemetry::trace::KeyValueIterableView<M> view{m1};
+
+	srand(time(0));
+
+	for (int i = 0; i < iterations; ++i)
+	{
+		uint8_t buf[16];
+		for (int j = 0; j < 16; ++j)
```

Comment:
You can use [GenerateRandomBuffer](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/test/common/random_test.cc#L18) in this case.